File: app/src/main/java/trunk/doi/base/ui/activity/SettingActivity.java
package trunk.doi.base.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.OnClick;
import trunk.doi.base.R;
import trunk.doi.base.base.BaseActivity;

import trunk.doi.base.dialog.ApkDownDialog;
import trunk.doi.base.util.ToastUtil;
import trunk.doi.base.view.TitleView;


public class SettingActivity extends BaseActivity {


    @BindView(R.id.tv_title)
    TitleView tvTitle;
    @BindView(R.id.setting_feedback_tv)
    TextView settingFeedbackTv;
    @BindView(R.id.update_version_num_tv)
    TextView updateVersionNumTv;

    @Override
    protected int initLayoutId() {
        return R.layout.activity_setting;
    }

    @Override
    protected void initView(@Nullable Bundle savedInstanceState) {


    }

    @Override
    protected void setListener() {

    }

    @Override
    protected void initData() {

    }

    @Override
    public void onBackPressed() {
        finish();
    }

    @OnClick({R.id.setting_feedback_tv, R.id.setting_recommend_friends_tv, R.id.setting_service_terms_tv, R.id.setting_about_us_tv, R.id.setting_cache_size_tv, R.id.setting_check_update_rl, R.id.setting_exit_login_btn})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.setting_feedback_tv:
                break;
            case R.id.setting_recommend_friends_tv:
                break;
            case R.id.setting_service_terms_tv:
                break;
            case R.id.setting_about_us_tv:
                break;
            case R.id.setting_cache_size_tv:
                break;
            case R.id.setting_check_update_rl:


                break;
            case R.id.setting_exit_login_btn:


                break;
        }
    }
}

File: app/src/main/java/trunk/doi/base/ui/activity/extra/SettingActivity.kt
package trunk.doi.base.ui.activity.extra

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import trunk.doi.base.R

/**
 * Created by ly on 2016/6/22.
 * 设置
 */
class SettingActivity : AppCompatActivity() {


    //  kotlin使用var，val关键字定义变量，如果在定义变量时直接赋值，则可以不用指定变量类型，否则需要在变量名后使用“：”来指定类型。
    // 在类型后加“？”表示该变量可空。如果变量可空，那在后续使用变量时，就必须先判断是否为空。
//    var a:Int = null  //错误，类型后没？，即不可空类型，自然不能等于null
//    var a:Int? = null  //正确
//
//    val canNull: Int ? = 0
//    canNull.toFloat  //错误，可空变量调用时要判断空
//    canNull?.toFloat   //正确，(如果是null，什么都不做)
//    canNull!!.Float   //正确，(如果null，报空指针)
//    val canNotNull: Int  = 0
//    canNotNull.toFloat  //正确

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)

    }

}
